std/experimental/allocator/package.d

```diff
@@ -560,6 +560,179 @@ unittest
     test(theAllocator);
 }
 
+// the class repository used in factory()
+package TypeInfo_Class[string] _factoryClasses;
+
+/**
+Registers a class to be available in the $(LREF factory) function.
```

Comment:
I think you should either make an explicit note in the documentation that the class registry is thread-local, or you should mark the hash table above as `shared`.  Most users coming from other languages will probably expect that the registry is global.